src/benchmarklib/table_builder.hpp

```diff
@@ -0,0 +1,240 @@
+#pragma once
+
+#include <optional>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+#include "boost/hana/assert.hpp"
+#include "boost/hana/for_each.hpp"
+#include "boost/hana/tuple.hpp"
+#include "boost/hana/zip_with.hpp"
+
+#include "resolve_type.hpp"
+#include "types.hpp"
+
+namespace hana = boost::hana;
+
+// helper structs and functions for compile time, the actual code is in namespace opossum
+namespace {
+// similar to std::optional but has_value is known at compile time, so "if constexpr" can be used
+// hana::optional does not allow moving and reinitializing its value (or I just did not find out how)
+template <typename T, bool _has_value, typename Enable = void>
+class optional_constexpr {
```

Comment:
I'm not sure we're consistent with this (we're not, see, `pmr_string`), but generally, class names should be camel case.